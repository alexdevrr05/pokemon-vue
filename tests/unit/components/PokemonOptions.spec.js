import { shallowMount } from '@vue/test-utils';
import { randomPokemons } from '../mocks/pokemons.mock';

import PokemonOptions from '@/components/PokemonOptions';

describe('PokemonOptions component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemonOptions: randomPokemons,
      },
    });
  });

  // toMatchInlineSnapshot genera un template string aquÃ­ como en __snapshots
  test('Debe de hacer match con el snapshot', () => {
    expect(wrapper.html()).toMatchInlineSnapshot();
  });

  // test('should ', () => {
  //   const wrapper = shallowMount(PokemonOptions, {
  //     props: {
  //       pokemonOptions: [],
  //     },
  //   });

  //   wrapper.vm.$emit('handleClickPokemonOption');
  //   expect(wrapper.emitted().handleClickPokemonOption).toBeTruthy();
  // });
});
