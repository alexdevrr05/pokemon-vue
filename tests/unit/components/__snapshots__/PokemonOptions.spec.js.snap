// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions component Debe de hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul id="example">
    <li>Bonsly</li>
    <li>Pikachu</li>
    <li>Sharpedo</li>
    <li>Misdreavus</li>
  </ul>
</div>
`;
